### PR TITLE
Mongo Connector examples [HZ-2296]

### PR DESCRIPTION
**Pull request by TomaszGaweda**
1. I've removed old example, which demonstrated how to write mapLoader with Mongo. Now it should be done via GenericMapStore
2. Implemented simple streaming pipeline with some aggregation to show the power of the dar... I mean, of the Jet engine.
3. Implemented simple SQL queries to show the SQL running

### CODE REVIEW

**Review by frant-hartm — APPROVED** [review]

File: jet/mongodb/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        <groupId>com.hazelcast.samples.jet</groupId>
+        <version>0.1-SNAPSHOT</version>
+    </parent>
+
+    <properties>
+        <!-- needed for checkstyle/findbugs -->
+        <main.basedir>${project.parent.parent.basedir}</main.basedir>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <java-version>17</java-version>
```
Comment (ldziedziul) [inline jet/mongodb/pom.xml:39]: Are we ok with samples requiring JDK17?

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * </ol>
+ *
+ */
+public class MongoSQLExample {
+
+    public static void main(String[] args) {
+        if (args.length > 0) {
+            new MongoSQLExample().run(args[0]);
+        } else {
+            try (MongoDBContainer mongoContainer = new MongoDBContainer("mongo:latest")) {
```
Comment (ldziedziul) [inline jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java:60]: Nice :)

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSourceExample.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                .stream(() -> MongoClients.create(connectionString))
+                .database("shop")
+                .filter(eq("fullDocument.successful", true))
+                .collection("payments", Payment.class)
+                .throwOnNonExisting(false)
+                .startAtOperationTime(new BsonTimestamp())
+                .build();
+        pipeline.readFrom(streamSource)
+                .withIngestionTimestamps()
+
```
Comment (ldziedziul) [inline jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSourceExample.java:105]:
```suggestion
```

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                    where successful = true
+                 ), descriptor(wallTime), interval '2' second))
+                 group by window_end, city
+                """)) {
+            print(result);
+        }
+    }
+
+    /**
+     * Pretty print (almost) any SqlResult.
```
Comment (ldziedziul) [inline jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java:114]: almost ;) 

image

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+/**
+ * Simple example that continuously reads from Mongo using SQL,
+ * prints average amount of payments for each city each 2 seconds.
+ *
+ * <p>
+ * You can run this example using one of two modes:
+ * <ol>
+ *      <li> Before running this example you can start a MongoDB instance, e.g. using Docker. Remember to configure
+ *          replica sets, as streaming queries replicas! In this mode you should manually compose the {@code connectionString}
+ *          using your Mongo instance coordinates and pass it as first argument of this example.
```
[the remark this answers is not in the document]
Comment (TomaszGaweda) [reply jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java:45]: I've checked how TestContainers do it and was like: "oh, f.. it, would be easier to use TestContainers if someone doesn't already have such replica set' :D But sure, I can prepare it

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+                    amount decimal external name "fullDocument.amount",
+                    successful boolean external name "fullDocument.successful",
+                    wallTime timestamp
+                )
+                connector type MongoStream
+                options (
+                    'connectionString' = '%s',
+                    'startAt' = '%s'
+                )
+                """.formatted(connectionString, startAt)).close();
```
[the remark this answers is not in the document]
Comment (TomaszGaweda) [reply jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java:99]: I didn't want to shout :( ;P

File: jet/mongodb/pom.xml (hunk 1 above)
Comment (TomaszGaweda) [reply jet/mongodb/pom.xml:39]: Yes, we already did it for CDC

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (TomaszGaweda) [reply jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java]: Ok, more compact version: daa519515a1253d2e8c0a51e7ac40c1d7cea0c11

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (TomaszGaweda) [reply jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java]: 837f6b8842562aa9bcb25543490819c52dde24c6

File: jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java (hunk 4 above)
Comment (TomaszGaweda) [reply jet/mongodb/src/main/java/com/hazelcast/samples/jet/mongodb/MongoSQLExample.java:121]: 837f6b8842562aa9bcb25543490819c52dde24c6

**Pull request merged**